rom/dev/src/flow/cold_reset/dice.rs

```diff
@@ -29,6 +29,8 @@ pub struct DiceInput<'a> {
 
     /// Authority Key Identifier
     pub auth_key_id: &'a [u8; 20],
+
+    pub fw_proc_info: FwProcInfo,
```

Comment:
A problem with having this field here is, this struct is passed to every DICE layer, even though it's only actually needed by `FmcAliasLayer::derive()`. Makes the IDevID and LDevID layers harder to reason about.

The `DiceLayer` trait is forcing that all inputs for any layer be added to the `DiceInput` struct, even though they aren't actually needed by all layers. Would suggest deleting the `DiceLayer` trait, and modifying `FmcAliasLayer::derive()` to take a `DiceInput` and a `FwProcInfo` argument, separately.

Incidentally, could then also modify `InitDevIdLayer::derive()` to not take any `DiceInput`, and delete `DiceInput::default()`. The only reason `DiceInput::default()` exists is because all layers are required to take a `DiceInput`, even if `InitDevIdLayer` doesn't need it.